daml/splice-util-featured-app-proxies/daml/Splice/Util/FeaturedApp/DelegateProxy.daml

```diff
@@ -0,0 +1,116 @@
+-- | Utility template for proxying token standard choices so that featured app markers
+-- are created when using extra parties for operating an app.
+--
+-- You can either use this template directly, or copy the code
+-- under a **different package name** and a **different module name**.
+--
+module Splice.Util.FeaturedApp.DelegateProxy where
+
+import Splice.Api.Token.TransferInstructionV1
+import Splice.Api.Token.AllocationV1
+import Splice.Api.Token.AllocationInstructionV1
+import Splice.Api.FeaturedAppRightV1
+
+-- | Generic argument to the proxy choices.
+data ProxyArg arg = ProxyArg with
+    choiceArg : arg -- ^ The argument to the choice that the delegate is exercising.
+    featuredAppRightCid : ContractId FeaturedAppRight -- ^ The featured app right contract of the provider.
+    beneficiaries : [AppRewardBeneficiary] -- ^ The beneficiaries for the activity marker.
+  deriving (Show, Eq)
+
+-- | Generic result of the proxy choices.
+data ProxyResult r = ProxyResult with
+    markerResult : FeaturedAppRight_CreateActivityMarkerResult
+    choiceResult : r
+
+-- | A proxy for a delegate to create featured app markers jointly with using token standard workflows.
+-- The delegate is given full control over the creation of the markers.
+--
+-- This is for example useful when using one or more parties for app operation purposes.
+-- Using this proxy allows the operational parties to properly attribute their activity to the featured app,
+-- which is represented by the app provider party.
+template DelegateProxy with
+    provider : Party -- ^ The app provider whose featured app right should be used
+    delegate : Party -- ^ The delegate interacting with the token standard workflow
```

Comment:
iirc for wallets we recommend to have a single delegate contract instead of one per delegate (which would be one per wallet user in those cases). Is there a good reason to do things differently here? I see why this is sufficient for exchanges but having to maintain both proxy variants feels unfortunate.

Reply:
If we do make the rules variant of that then we likely want to also consider whether there should be a way to restrict beneficiaries, I don't expect that giving full control over that to users is always desired.

Maybe a cheap option to start with would be to just have a flag on the rules contract that defines whether you either use a set of beneficiaries specified on the contract or let users chose it.

Reply:
I initially intended to go in that direction of making a single highly configurable version of this TokenStandardProxy here: https://github.com/hyperledger-labs/splice/pull/1907/files#diff-4cbc6e851f73f40db384d63aa97dbf4ffb93c505b0ed0c3c360e7914f94f6201R42

However, it felt uneccessarily complicated compared to the option where we just define two different templates in different modules of the same utility package. This way the user just chooses what template to use and goes with it.

Note that the difference is not only in the template arguments, but also the proxy argument. 

Furthermore, I wouldn't be surprised if we find that we need a version 2 of one of the proxies, and we can just add that to the side w/o impacting existing users.

Reply:
makes sense, maybe make it clear somehow that this is not the recommended template for wallets and link to your PR for that for now. I suspect people will get confused otherwise.

Reply:
https://github.com/hyperledger-labs/splice/issues/2197